### PR TITLE
Add preprocessor unit test

### PR DESCRIPTION
**Pull request by XiaofeiCao**
Fix https://github.com/Azure/autorest.java/issues/1256

Add a test case for preprocessing code model file.

### CODE REVIEW

**Comment by weidongxu-microsoft** [comment]
You will need a surefire plug to run the test in `mvn verify`
https://github.com/Azure/autorest.java/blob/main/fluentgen/pom.xml#L95-L111

File: preprocessor/src/test/java/com/azure/autorest/PreprocessorTests.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        // SOURCE SWAGGER URL: https://github.com/Azure/azure-rest-api-specs/blob/main/specification/containerregistry/data-plane/Azure.ContainerRegistry/stable/2021-07-01/containerregistry.json
+        // It's a debug file from the output of modelerfour during the generate process where it converts swagger definition into code model.
+        // This file can be obtained by calling generate commands from `generate` or `generate.bat` on the swagger json, it will appear under this project by the name 'code-model.yaml'.
+        String codeModelFileName = "containerregistry-code-model.yaml";
+
+        CodeModel codeModel = preprocessor.loadCodeModel(codeModelFileName);
+        codeModel = preprocessor.transform(codeModel);
+        String output = preprocessor.dump(codeModel);
+
+        System.out.println(output);
```
Comment (weidongxu-microsoft) [inline preprocessor/src/test/java/com/azure/autorest/PreprocessorTests.java:30]: Add some assertion e.g. about operation group, operation, model in next PR.

**Review by weidongxu-microsoft — APPROVED** [review]

File: preprocessor/src/test/java/com/azure/autorest/MockPreprocessor.java [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+    private static final Map<String, Object> SETTINGS_MAP = new HashMap<>();
+
+    static {
+        SETTINGS_MAP.put("generate-client-interfaces", false);
+        SETTINGS_MAP.put("generate-client-as-impl", true);
+        SETTINGS_MAP.put("generate-sync-async-clients", true);
+        SETTINGS_MAP.put("add-context-parameter", true);
+        SETTINGS_MAP.put("context-client-method-parameter", true);
+        SETTINGS_MAP.put("sync-methods", "all");
```
Comment (weidongxu-microsoft) [inline preprocessor/src/test/java/com/azure/autorest/MockPreprocessor.java:42]: Do we need all these in preprocessor test?

Comment (XiaofeiCao) [reply preprocessor/src/test/java/com/azure/autorest/MockPreprocessor.java:42]: Maybe... not... I copied it from readme under preprocessor. I removed it and rerun the test to find it pass. 
Will remove it in the next pr.

File: preprocessor/src/test/java/com/azure/autorest/PreprocessorTests.java (hunk 1 above)
Comment (XiaofeiCao) [reply preprocessor/src/test/java/com/azure/autorest/PreprocessorTests.java:30]: Will do.

**Pull request merged**